src/hdfs.rs

```diff
@@ -647,23 +647,27 @@ impl HdfsFile {
 
     /// Write data into an open file.
     pub fn write(&self, buf: &[u8]) -> Result<i32, HdfsErr> {
```

Comment:
To align with the semantics of LocalFileSystem write, and imitate the behavior of https://doc.rust-lang.org/src/std/io/mod.rs.html#1540-1555, which actually allows an empty buffer as input, but writes the whole buffer out.

Reply:
As https://github.com/datafusion-contrib/fs-hdfs/blob/cadaee54e20f9c09484dae088485908ece966893/c_src/libhdfs/hdfs.c#L1464-L1466 mentioned, the `hdfsWrite` won't write partial contents. Maybe it's better to refine the condition check as follows:
`
if written_len >= 0 {
            Ok(written_len)
        } else {
            Err(HdfsErr::Generic(format!(
                "Fail to write contents to file {}",
                self.path()
            )))
        }
`

Reply:
Thanks, will do

Reply:
Reversed, only allow to call with an empty buffer and do nothing for this case.